### PR TITLE
Add long explanation for E0726

### PR DESCRIPTION
**Pull request by codekidX**
Hey, I think some refactoring of codebase has occurred so the guidance mentioned in  #61137 is not updated I think @GuillaumeGomez .

Please let me know if any other changes needs to be added to this PR.

### CODE REVIEW

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+A lifetime indication has not been added to specify the span of your type.
+
+Erroneous code example:
+
+```rust
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:5]:
``````suggestion
```compile_fail,E0725
``````

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+struct SomeStruct;
+
+impl SomeStruct {
+    pub fn some_fn(&self, abc: Abc) { // error: implicit elided lifetime not allowed here rustc(E0726)
+        // ...
+    } 
+}
+```
+
+Specify the lifetime of struct `Abc` or indicate the anonymous lifetime like `abc: Abc<'_>`.
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:19]: The explanation is a bit lacking no? Maybe try to explain a bit more.

Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/error_codes/E0726.md:19]: Also, please add a working version of the code example from above.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
The explanation needs to be extending (lifetimes are far from an easy topic in rust) and you also need to add a working version of your previous failing code example.

File: compiler/rustc_error_codes/src/error_codes.rs [hunk 3]
```diff
@@ -423,6 +423,7 @@ E0720: include_str!("./error_codes/E0720.md"),
 E0722: include_str!("./error_codes/E0722.md"),
 E0724: include_str!("./error_codes/E0724.md"),
 E0725: include_str!("./error_codes/E0725.md"),
+E0725: include_str!("./error_codes/E0726.md"),
 E0727: include_str!("./error_codes/E0727.md"),
 E0728: include_str!("./error_codes/E0728.md"),
 E0729: include_str!("./error_codes/E0729.md"),
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes.rs:429]: You need to remove E0725 from the list of error codes from below (it's not supposed to be declared twice in this file).

File: compiler/rustc_error_codes/src/error_codes.rs [hunk 4]
```diff
@@ -423,6 +423,7 @@ E0720: include_str!("./error_codes/E0720.md"),
 E0722: include_str!("./error_codes/E0722.md"),
 E0724: include_str!("./error_codes/E0724.md"),
 E0725: include_str!("./error_codes/E0725.md"),
+E0725: include_str!("./error_codes/E0726.md"),
```
Comment (Urgau) [inline compiler/rustc_error_codes/src/error_codes.rs:426]: It's also *726* and not *725*.
```suggestion
E0726: include_str!("./error_codes/E0726.md"),
```

**Comment by codekidX** [comment]
@GuillaumeGomez does `E0726` it not have a test case? if not, where should I add it?

**Comment by codekidX** [comment]
hey @GuillaumeGomez, I have added detailed explanation, please take a look.

Also the test is failing as
```
Some errors have detailed explanations: E0277, E0726.
For more information about an error, try `rustc --explain E0277`.
```

it should be 
```
Some errors have detailed explanations: E0726.
```
right?

Since its itself `E0277`. Do let me know if this is an issue or I need to add E0277 above as well.

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+When a struct or a type is bound/declared with a lifetime it is important for
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:1]: Please write a short summary of the error here, the explanation goes below. :)

Something like "An argument's lifetime was elided in an async function".

**Comment by GuillaumeGomez** [comment]
> @GuillaumeGomez does `E0726` it not have a test case? if not, where should I add it?

In `src/test/ui/error_codes/` (there are a few others in this folder too so you can check how it works).

For the failing ui test, simply run it with `--bless` and let the tester do its job. ;)

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+struct Content<'a> {
+    title: &'a str,
+    body: &'a str,
+}
+
+struct Blog;
+
+impl Blog {
+    pub async fn create(content: Content) { // error: implicit elided
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:19]: Why do you need a method btw? Shouldn't an async function be enough?

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                                            // lifetime not allowed here
+        println!("title: {}", content.title);
+        println!("body: {}", content.body);
+    }
+}
+
+fn main() {
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:26]: No need to declare a `main` function, please remove it (but keep its content!).

File: compiler/rustc_error_codes/src/error_codes/E0726.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+Specify desired lifetime of parameter `content` or indicate the anonymous
+lifetime like `content: Content<'_>`. The anonymous lifetime tells the Rust
+compiler that `content` is only needed until create function is done with
+it's execution.
+
+The `implicit elision` meaning the omission of suggested lifetime that is
+`pub async fn create<'a>(content: Content<'a>) {}` is not allowed here as
+lifetime of the `content` can differ from current context.
+
+Know more about lifetime elision in this
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0726.md:42]: Please add a working version of your example above.

**Comment by JohnCSimon** [comment]
Ping from triage: 
@codekidX can you please address the merge conflicts and build failures. Thank you.

**Comment by JohnCSimon** [comment]
Ping again from triage:
@codekidX can you please address the merge conflicts and build failures. Thank you.

**Comment by codekidX** [comment]
@JohnCSimon I'll rebase and resolve comments by today. Had some busy weeks, thanks for waiting.

**Comment by jyn514** [comment]
Hi @apiraino, why did you add T-rustdoc here? This is a compiler error, rustdoc doesn't introduce any error codes that I know of.

**Comment by apiraino** [comment]
> why did you add T-rustdoc here? This is a compiler error, rustdoc doesn't introduce any error codes that I know of.

@jyn514 I sweep the repository and assign the relevant `T-*` labels. Here I have assigned the wrong one.

I probably didn't have clear the context and having seen a review from Guillaume tricked me into assuming that t-rustdoc had a say on this. Sorry, I hope that didn't cause any issues!

**Comment by jyn514** [comment]
@apiraino no worries, I changed it to the right label :)

**Comment by jyn514** [comment]
Hi @codekidX, do you know how to fix the CI error? I think you just need to pass `--bless` to the tests :) feel free to ask me here or [on Zulip](https://rust-lang.zulipchat.com/#narrow/pm-with/232545-user232545) if you're having trouble!

**Comment by ThrashAbaddon** [comment]
bump!

**Comment by JohnCSimon** [comment]
@codekidX 
Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**